### ISSUE TITLE
Process.Start() is stuck on Cent OS 7 and Ubuntu 18.04

### ISSUE THREAD

**Issue by anatolybolshakov** [dotnet/core#5717]
# Issue Title

Process.Start() is stuck sometimes on Cent OS 7 and Ubuntu 18.04

# General

Recently we updated Net core version used by Azure pipelines agent to 3.1.0, and it seems that for some customers issue appeared - Process.Start("chmod 600 <file in agent folder>") is stuck for some reason (on [line](https://github.com/microsoft/azure-pipelines-agent/blob/master/src/Agent.Sdk/ProcessInvoker.cs#L285))
At the same moment calling this command from terminal works fine for customer. Agent process is being run under the root, nothing else changed in code for this call.

Here are Process.StartInfo parameters for call:
File name: '/bin/chmod'
Arguments: '600 <file in agent folder>'
Working directory: '/root/azagent'
UseShellExecute: false
CreateNoWindow: true
RedirectStandardInput: true;
RedirectStandardError: true;
RedirectStandardOutput: true

OS version: Cent OS 7 (centos-release-7-6.1810.2.el7.centos.x86_64), Ubuntu 18.04
Net core version: 3.1.0

According to logs it seems that chmod process is in zombie state after the call, and there's no return value from Process.Start() method:

https://github.com/microsoft/azure-pipelines-agent/issues/3093#issuecomment-721630338

You can also find more info in thread for [issue](https://github.com/microsoft/azure-pipelines-agent/issues/3093).

Could you please take a look? Do you know if there were any similar issues - if so, is there any possible workaround/is it fixed?

**Comment by scalablecory** [comment]
I see you're redirecting stdin here -- chmod may be expecting you to close that before it can complete. Generally don't redirect the streams unless you intend to read/write them.

**Comment by anatolybolshakov** [comment]
@scalablecory thanks! We will try to disable redirect for the case when there's no input indented.

**Comment by wfurt** [comment]
cc: @tmds

**Comment by tmds** [comment]
> According to logs it seems that chmod process is in zombie state after the call, and there's no return value from Process.Start() 

Can you get the stacktraces from all threads? That will give us some information about where `Process.Start` is blocked, and maybe what other thread is blocking it.

> Generally don't redirect the streams unless you intend to read/write them.

I'd generally recommend to redirect streams of apps you don't want the user to interact with directly.  After redirecting, you should close the input stream, and consume the output/error streams.

**Comment by tmds** [comment]
> Net core version: 3.1.0

Can you also try using the latest 3.1 patch version?

**Comment by anatolybolshakov** [comment]
@tmds yes thanks, I will increase logs verbosity and we will try it with a customer also
